src/main/java/org/spongepowered/common/world/volume/buffer/archetype/AbstractReferentArchetypeVolume.java

```diff
@@ -303,7 +303,7 @@ public int highestYAt(final int x, final int z) {
                         // Order of operations matters here, block states need to be mirrored first, then rotated
                         .mirror(mirror)
                         .rotate(this.transformation.rotation()),
-                    this.transformStreamBlockPosition(e.position())
+                    this.transformStreamBlockPosition(e.position().add(VolumePositionTranslators.BLOCK_OFFSET)).sub(VolumePositionTranslators.BLOCK_OFFSET)
```

Comment:
@gabizou This was the fix. If you look at `transformStreamBlockPosition`, the comment says:

```
        // we assume that the position is already center adjusted, but since
        // we're not going to want to "flatten" the positions, we need to correctly
        // round and then re-add the offset post transformation
```

It wasn't.

This might not be the best place to fix it, but I'm not sure where it should go in the grand scheme of things.